### ISSUE TITLE
Unable to create VLAN

### ISSUE THREAD

**Issue by andreasbaekdahl** [core-infra-svcs/terraform-provider-meraki#241]
It is not possible to create a new meraki_networks_appliance_vlans resource.

```
resource "meraki_networks_appliance_vlans" "test" {
    network_id = "L_645140639920850511"
    appliance_ip = "10.10.10.1"
    subnet = "10.10.10.0/24"
    vlan_id = 10
    name = "Office"
}

```

```
> terraform apply

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  # meraki_networks_appliance_vlans.test will be created
  + resource "meraki_networks_appliance_vlans" "test" {
      + appliance_ip              = "10.10.10.1"
      + cidr                      = (known after apply)
      + dhcp_boot_options_enabled = (known after apply)
      + dhcp_handling             = (known after apply)
      + dhcp_lease_time           = (known after apply)
      + dhcp_options              = (known after apply)
      + dns_nameservers           = (known after apply)
      + id                        = (known after apply)
      + mask                      = (known after apply)
      + name                      = "Office"
      + network_id                = "L_645140639920850511"
      + reserved_ip_ranges        = (known after apply)
      + subnet                    = "10.10.10.0/24"
      + template_vlan_type        = (known after apply)
      + vlan_id                   = 10
    }

Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

meraki_networks_appliance_vlans.test: Creating...
2023-11-15T22:29:33.121+0100 [ERROR] provider.terraform-provider-meraki_v0.5.1: Response contains error diagnostic: diagnostic_detail="An unexpected error was encountered trying to build a value. This is always an error in the provider. Please report the following to the provider developer:

Received unknown value, however the target type cannot handle unknown values. Use the corresponding `types` package type or a custom type that handles unknown values.

Path: dhcp_options
Target Type: []provider.NetworksApplianceVLANsResourceModelDhcpOption
Suggested Type: basetypes.SetValue" tf_provider_addr=registry.terraform.io/core-infra-svcs/terraform-provider-meraki @caller=github.com/hashicorp/terraform-plugin-go@v0.18.0/tfprotov6/internal/diag/diagnostics.go:58 @module=sdk.proto tf_proto_version=6.3 tf_rpc=ApplyResourceChange diagnostic_attribute=AttributeName("dhcp_options") diagnostic_severity=ERROR diagnostic_summary="Value Conversion Error" tf_req_id=7ffed692-fed3-31f1-6578-f00118414992 tf_resource_type=meraki_networks_appliance_vlans timestamp=2023-11-15T22:29:33.121+0100
2023-11-15T22:29:33.129+0100 [ERROR] vertex "meraki_networks_appliance_vlans.test" error: Value Conversion Error
╷
│ Error: Value Conversion Error
│ 
│   with meraki_networks_appliance_vlans.test,
│   on networks.tf line 29, in resource "meraki_networks_appliance_vlans" "test":
│   29: resource "meraki_networks_appliance_vlans" "test" {
│ 
│ An unexpected error was encountered trying to build a value. This is always an error in the provider. Please report the following to the provider
│ developer:
│ 
│ Received unknown value, however the target type cannot handle unknown values. Use the corresponding `types` package type or a custom type that handles
│ unknown values.
│ 
│ Path: dhcp_options
│ Target Type: []provider.NetworksApplianceVLANsResourceModelDhcpOption
│ Suggested Type: basetypes.SetValue
```

In the example above, the path dhcp_options is failing. If I add an empty value for that in the resource argument, it leads to the next optional argument failing: reserved_ip_ranges, fixed_ip_assignments, ipv6, mandatory_dhcp, dhcp_options, template_vlan_type

If all these optional arguments are filled out with values, the final error is:

```
2023-11-15T22:33:14.891+0100 [ERROR] vertex "meraki_networks_appliance_vlans.test" error: Provider returned invalid result object after apply
2023-11-15T22:33:14.891+0100 [ERROR] vertex "meraki_networks_appliance_vlans.test" error: Provider returned invalid result object after apply
2023-11-15T22:33:14.891+0100 [ERROR] vertex "meraki_networks_appliance_vlans.test" error: Provider returned invalid result object after apply
```

**Comment by iamdexterpark** [comment]
Hello @andreasbaekdahl 

Thank you for pointing this out! We are seeing an issue with native go-types within some of our resources and are in the process of refactoring. 

I've started with this resource and will try to fix it as soon as possible.

**Comment by matthijsf** [comment]
I'm seeing the same issue with vlan creation.

**Comment by abelal83** [comment]
@iamdexterpark just checking to see if any progress was made?

**Comment by iamdexterpark** [comment]
I appreciate your patience; we have pushed a fix in v0.6.0 to address this issue. Please upgrade to the latest version and let us know if you encounter any other issues.
